### ISSUE TITLE
How to delete an existing header using plugins?

### ISSUE THREAD

**Issue by lucas-santosP** [ecyrbe/zodios#280]
### The problem
I tried using the `pluginHeader` and `pluginToken`, my objective was to remove the "Authorization" header because for example using Firebase, if exist the "Authorization" header (even if is undefined) it will try to validate it and if fails will throw an error. That includes the login endpoint which ideally should be called without auth.

So my problem is: I want to use the Authorization header for most of my API calls, but some specific ones like the login endpoint should not have the header.

### Solution suggestion

I looked into the plugin's code and I saw that the `pluginToken` with `getToken` and `renewToken` ignores undefined values, and the `pluginHeader` did not accept an undefined value. I suggest changing this behavior (in both plugins) to consider undefined as a way to delete an existing header.

In resume:
- If the header did not exists and undefined is received, nothing is done.
- If the header exists and undefined is received, the header is deleted.

Syntax suggestion:
```ts
// set the Authorization header to all endpoints
apiClient.use(
  pluginToken({ getToken: async () => "auth-token" })
);

// delete the Authorization header for the login endpoint using pluginToken
apiClient.use("login", pluginToken({ getToken: async () => undefined }));
// doing the same as above but with pluginHeader
apiClient.use("login", pluginHeader("Authorization", async () => undefined));
```

If you think this is valid I can create a PR doing it. but if this undefined value behavior cannot be changed for any reason we could think of another way to enable the deletion, maybe adding another function for that.

**Comment by ecyrbe** [comment]
Hello, thank you for the suggestion.
Maybe a plugin that cleans the header explicitly would be better to not generate unexpected side effects on existing users of the plugins.
So maybe something like this could be more appropriate :
```ts
apiClient.use("login", pluginClearHeader("Authorization"));
```

**Comment by lucas-santosP** [comment]
I see... changing the `pluginToken` really can be a problem to change for existing users since it already accepts undefined. But the `pluginHeader` did not accepts undefined values in its type definition so it might not cause a side effect right? Maybe only for js users...

But anyway, if you still prefer the creation of another plugin to enable deletion, I'm willing to do that too. Then I just would suggest later (in a bigger update) the renaming of the `pluginHeader` to `pluginSetHeader` to fit better together with the `pluginClearHeader`.

**Comment by ecyrbe** [comment]
yes, i was thinking this exacly while suggesting it. I'll definitly rename it for v11

**Comment by lucas-santosP** [comment]
Great, I'll open the PR for `pluginClearHeader` soon.

**Comment by ecyrbe** [comment]
Thank you for the PR, this is available in in `@zodios/plugins` `v10.5.0`
